python/ray/air/_internal/remote_storage.py

```diff
@@ -199,22 +207,38 @@ def _translate_s3_options(options: Dict[str, List[str]]) -> Dict[str, Any]:
 
     """
     # Map from s3 query keys --> botocore client arguments
+    # client_kwargs
     option_map = {
         "endpoint_override": "endpoint_url",
         "region": "region_name",
         "access_key": "aws_access_key_id",
         "secret_key": "aws_secret_access_key",
     }
+    client_kwargs = _translate_options(option_map, options)
 
-    client_kwargs = {}
-    for opt, target in option_map.items():
-        if opt in options:
-            client_kwargs[target] = options[opt][0]
+    # config_kwargs
+    option_map = {
+        "signature_version": "signature_version",
+    }
+    config_kwargs = _translate_options(option_map, options)
+
+    # s3_additional_kwargs
+    option_map = {
+        "ServerSideEncryption": "ServerSideEncryption",
+        "SSEKMSKeyId": "SSEKMSKeyId",
+        "GrantFullControl": "GrantFullControl",
```

Comment:
It's hard to come up with a dynamic approach to cover all possible options here without churning the APIs too much. I'd like to revisit this when we think about custom upload logic and Syncer deprecation, and for now go with manually selected properties to forward.

Reply:
Yeah, I agree that we should revisit this. I was searching for other possibilities and `ACL` is another param that can be passed into `s3_additional_kwargs`. Then, GCS also has a bunch of query params that users can specify.

Another thing to note: with the current `Syncer` abstraction, users could technically just create an s3fs client and customize all of these settings there.

Reply:
Oh, I see, the custom `Syncer` doesn't affect `Checkpoint.to_uri` at all.